### ISSUE TITLE
[merp] SendMicrosoftTelemetry icall doesn't do any MERP logging when inducing a Managed Crash

### ISSUE THREAD

**Issue by BenBeckley** [mono/mono#17790]
## Steps to Reproduce

Induce a "Managed Crash" from a debug build of VSMac (Help > Diagnostics > Induce Managed Crash)

### Current Behavior

VSMac crashes, and despite the MERP dialog appearing, there are no breadcrumbs left in the VSMac log directory.

### Expected Behavior

Breadcrumbs should be left in the VSMac log directory indicating that MERP logging is working properly.

## On which platforms did you notice this

[x] macOS
[ ] Linux
[ ] Windows

**Version Used**:

Mono Framework MDK:
Mono 6.6.0.140 (2019-08/1b2e536b223) (64-bit)

VSMac built from:
xamarin/md-addins master/2e1d72e25e539510ab8296d111b874b3f3048cb5
mono/monodevelop master/222fcf249af551c9b8b43ada07daf677ede74ab7
## Stacktrace

```
FATAL ERROR [2019-11-05 13:31:23Z]: An unhandled exception has occurred. Terminating Visual Studio? True
System.Exception: Diagnostics: Managed crash
  at PerformanceDiagnosticsAddIn.InduceManagedCrashHandler+<>c.<Run>b__0_0 () [0x00001] in /Users/Ben/src/vs-mac/master/monodevelop/main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Commands.cs:179 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-08/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/thread.cs:74 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-08/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:968 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-08/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:910 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-08/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:899 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-08/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/thread.cs:111 
2019-11-05 13:31:23.689 MonoDevelop[56582:2039068] warning: exception inside UnhandledException handler: (null) assembly:/Library/Frameworks/Mono.framework/Versions/6.6.0/lib/mono/4.5/mscorlib.dll type:TargetInvocationException member:(null)
[ERROR] FATAL UNHANDLED EXCEPTION: System.Exception: Diagnostics: Managed crash
  at PerformanceDiagnosticsAddIn.InduceManagedCrashHandler+<>c.<Run>b__0_0 () [0x00001] in /Users/Ben/src/vs-mac/master/monodevelop/main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Commands.cs:179 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-08/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/thread.cs:74 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-08/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:968 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-08/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:910 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-08/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:899 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-08/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/thread.cs:111 
make[1]: *** [run-bundle] Error 1
make: *** [run] Error 2
```

**Comment by lambdageek** [comment]
Need to turn on the timeline for capturing managed exceptions, @alexischr

**Comment by marek-safar** [comment]
@alexischr what is update on this issue?

**Comment by BenBeckley** [comment]
Fixed two typos in the repro steps from "Induce UI Thread Crash" to "Induce Managed Crash".

**Comment by alexischr** [comment]
Going by the title of the issue, the `SendMicrosoftTelemetry` icall should be creating the breadcrumbs (as it calls `mono_merp_invoke`, which uses the timeline, and use of the timeline creates the breadcrumbs). I am now looking into whether or not everything on the VS4Mac side is fully wired up to use `Mono.Runtime.SendExceptionToTelemetry`, which is the right icall to use for this functionality (and also call `SendMicrosoftTelemetry`).

**Comment by alexischr** [comment]
After further investigation we found that because the timeline wasn't initialized properly, all further calls to progress the timeline and generate the breadcrumbs were not being really executed.

**Comment by lambdageek** [comment]
Reopening until the 2019-10 backport #17945  is merged.

**Comment by lambdageek** [comment]
Fixed on master and 2019-12 (#17913) and on 2019-10 (#17945) now.
